src/line_launcher/duration_slider_listener.rs

```diff
@@ -0,0 +1,35 @@
+use crate::{config, config::midi::MidiSlider, midi, Message};
+use std::sync::{mpsc::Receiver, Arc, Mutex};
+use std::thread;
+use wmidi::Channel;
+
+pub fn listen_for_duration_control_changes(
```

Comment:
I was trying to express this as a `DurationSliderListener` struct at first that registered itself with the `MidiMessagePublisher` to get callbacks when new MIDI messages come in and would then mutate its own internal "current duration fraction/percent" state in response, and then the note-off-instruction logic in `LineLauncher` would examine that state. I was running into lifetime/borrowing issues so did what seemed a little simpler where there's no struct encapsulating it, there's just a mutex-guarded float value (representing the duration percent) that gets passed in here along with a receiver of MIDI messages and it mutates the float value if it sees control messages corresponding to the `MidiSlider` that's currently hardcoded here